src/app/charts/line-graph.component.ts

```diff
@@ -96,20 +96,23 @@ export class LineGraphComponent {
 
     // Set axis scales
     private setAxisScales(): void {
-        var parseTime = D3.timeParse('%Y-%m-%d');
+        // Time scales only recognize annual and daily data
+        var parseTime = D3.timeParse('%Y'); //D3.timeParse('%Y-%m-%d');
         this.extractedData.forEach(d => d.date = parseTime(d.date));
         this.xRange = D3.extent(this.extractedData, d => d.date);
         this.xScale.domain(this.xRange);
-        this.yScale.domain([0, D3.max(this.yData)]);
+        // Adjust y scale, prettify graph
+        let yPad = Math.abs((D3.max(this.yData) - D3.min(this.yData)) * 2/3);
+        this.yScale.domain([D3.min(this.yData) - yPad, D3.max(this.yData) + yPad]);
     }
 
     /* Will draw the X Axis */
     private drawXAxis(): void {
         this.svg.append('g')
           .attr('transform', 'translate(0,' + this.height + ')')
           .call(D3.axisBottom(this.xScale)
-          .ticks(5)
-          .tickFormat(D3.timeFormat('%m-%Y')));
+          .ticks(3)
+          .tickFormat(D3.timeFormat('%Y')));
```

Comment:
Related to other date formatting comment: the formatting string for a given chart should probably go in a chart-dependent variable, so it can be both reused within a given chart and also modified for different aggregation levels across charts.